From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: delightfulcoffeecake <delightfulcoffeecake@proton.me>
Date: Mon, 20 Jul 2026 13:41:28 +0100
Subject: [PATCH] 9400: Build: Add aarch64 support to standalone decompiler
 Makefile

Build: Add aarch64 support to decompiler Makefile

remove extra whitespace
---
 Ghidra/Features/Decompiler/src/decompile/cpp/Makefile | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/Ghidra/Features/Decompiler/src/decompile/cpp/Makefile b/Ghidra/Features/Decompiler/src/decompile/cpp/Makefile
index 2d4eccabe9..7bfa99d742 100644
--- a/Ghidra/Features/Decompiler/src/decompile/cpp/Makefile
+++ b/Ghidra/Features/Decompiler/src/decompile/cpp/Makefile
@@ -12,7 +12,6 @@ GHIDRA_BIN=../../../../../../../ghidra.bin
 OS = $(shell uname -s)
 CPU = $(shell uname -m)
 
-# TODO: need to revise to support arm64/aarch64 arch - improve on both OS and arch detection 
 
 ifeq ($(OS),Linux)
 # Allow ARCH to be specified externally so we can build for 32-bit from a 64-bit Linux
@@ -22,11 +21,17 @@ endif
 ifeq ($(ARCH),x86_64)
   ARCH_TYPE=-m64
   OSDIR=linux_x86_64
+else
+ifeq ($(ARCH),aarch64)
+  # ARM64 compilers don't need an architecture flag for native builds
+  ARCH_TYPE=
+  OSDIR=linux_aarch64
 else
   ARCH_TYPE=-m32
   OSDIR=linux_x86_32
 endif
 endif
+endif
 
 ifeq ($(OS),Darwin)
   MAKE_STATIC=
-- 
2.45.1

